src/components/ebay-calendar/component.js

```diff
@@ -0,0 +1,316 @@
+            iso = offsetISO(iso, 1);
+        }
+        return iso;
+    }
+
+    getLastActiveISO(input = this.input) {
+        let iso = this.getLastVisibleISO(input);
+        const firstVisible = this.getFirstVisibleISO();
+        while (iso >= firstVisible && this.isDisabled(iso)) {
+            iso = offsetISO(iso, -1);
+        }
+        return iso;
+    }
+
+    /**
+     * @param {Date} date
+     */
+    monthTitle(date) {
+        const formatter = new Intl.DateTimeFormat(localeOverride(this.input.locale), {
+            month: 'long',
+            year: 'numeric',
+        });
+        return formatter.format(date);
+    }
+
+    /**
+     * @param {boolean | void} focus
+     */
+    prevMonth(focus) {
+        if (this.state.disableBefore && this.getFirstVisibleISO() <= this.state.disableBefore) {
+            return false;
+        }
+
+        this.state.offset--;
+        let newTabindexISO = this.state.tabindexISO;
+        const lastActiveISO = this.getLastActiveISO();
+        if (this.state.tabindexISO > lastActiveISO) {
+            newTabindexISO = this.state.tabindexISO = lastActiveISO;
+        }
+        if (focus) {
+            this.setTabindexAndFocus(newTabindexISO);
+        }
+
+        this.emit('month', { iso: toISO(this.getMonthDate(this.state.offset)) });
+
+        return true;
+    }
+
+    /**
+     * @param {boolean | void} focus
+     */
+    nextMonth(focus) {
+        if (this.state.disableAfter && this.getLastVisibleISO() >= this.state.disableAfter) {
+            return false;
+        }
+
+        this.state.offset++;
+        let newTabindexISO = this.state.tabindexISO;
+        const firstActiveISO = this.getFirstActiveISO();
+        if (this.state.tabindexISO < firstActiveISO) {
+            newTabindexISO = this.state.tabindexISO = firstActiveISO;
+        }
+        if (focus) {
+            this.setTabindexAndFocus(newTabindexISO);
+        }
+
+        this.emit('month', {
+            iso: toISO(this.getMonthDate(this.state.offset + (this.input.numMonths || 1))),
+        });
+
+        return true;
+    }
+
+    /**
+     * @param {DayISO} iso
+     */
+    setTabindexAndFocus(iso) {
+        this.state.tabindexISO = iso;
+        // After UI updates, focus on the new tabindex date
+        setTimeout(() => {
+            /** @type {HTMLButtonElement | undefined} */ (this.getEl(iso))?.focus();
+        });
+    }
+
+    /**
+     * @param {Input} input
+     */
+    calculateRangeDisplay(input = this.input) {
+        if (input.selected && input.range) {
+            // Determine range display (state.rangeStart-state.rangeEnd)
+            let iso1, iso2;
+            if (Array.isArray(input.selected)) {
+                [iso1, iso2] = input.selected;
+            } else if (this.state.focusISO) {
+                iso1 = input.selected;
+                iso2 = this.state.focusISO;
+            }
+            if (iso1 && iso2) {
```

Comment:
Some comments here to describe what is happening would be helpful.

Reply:
Sure, comments added